src/raft/raw_node.rs

```diff
@@ -300,6 +312,9 @@ impl<T: Storage> RawNode<T> {
            raft.raft_log.has_next_entries() {
             return true;
         }
+        if raft.read_state.index != INVALID_INDEX {
```

Comment:
The PR https://github.com/coreos/etcd/pull/6401 is held up by  https://github.com/coreos/etcd/pull/6275. We could port these PRs from etcd into TiKV all in once.